### ISSUE TITLE
[PWA] No manifest detected (UTF-8 BOM causes Stencil builder to corrupt the index.html file while injecting things in it)

### ISSUE THREAD

**Issue by JCKodel** [ionic-team/stencil#1272]
**Stencil version:**

```
 @stencil/core@0.16.0
```

**I'm submitting a:**

[X] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**
PWA doesn't work:
![image](https://user-images.githubusercontent.com/379339/49679982-a409f100-fa76-11e8-9ecb-ed5b63db4b19.png)

**Expected behavior:**
Manifest to load.

**Steps to reproduce:**
Created a project with PWA template.
Tried everything.
This is my current config:

```
serviceWorker:
{
    //swSrc: "src/sw.js",
    globPatterns: ["**/*.{js,css,json,html,ico,png,svg,xml}"],
    navigateFallback: "/",
    runtimeCaching: [
        { urlPattern: /[\s\S]*/, handler: "networkFirst" },

        {
            urlPattern: /^https:\/\/fonts\.googleapis\.com\//,
            handler: "cacheFirst",
            options:
            {
                cacheName: "google-fonts-cache", expiration: { maxEntries: 5, maxAgeSeconds: 6000 },
            }
        }
    ],
},
```

Tried with default sw.js. No lucky.

The manifest do exists (https://seleto.menu/manifest.json) so the sw.js:

![image](https://user-images.githubusercontent.com/379339/49680036-678ac500-fa77-11e8-8c4c-608631c52acf.png)

**Comment by JCKodel** [comment]
Found the cause of this bug (but not the cause of this cause =P):

My index.html is corrupted after build:

![image](https://user-images.githubusercontent.com/379339/49680353-b128df00-fa7a-11e8-902e-799d307c9463.png)

This is the original source in src folder:
```html
<!DOCTYPE html>
<html dir="ltr">
<head>
    <link rel="manifest" href="/manifest.json">
    <link href="/build/app.css" rel="stylesheet">
    <script src="/build/app.js"></script>
    <meta charset="utf-8">
    <title>Seleto Food</title>
    <meta name="viewport" content="viewport-fit=cover, width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <link rel="apple-touch-icon" sizes="180x180" href="/assets/icon/apple-touch-icon.png">
    <link rel="icon" type="image/png" sizes="192x192" href="/assets/icon/android-chrome-192x192.png">
    <link rel="icon" type="image/png" sizes="512x512" href="/assets/icon/android-chrome-512x512.png">
    <link rel="icon" type="image/png" sizes="32x32" href="/assets/icon/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/assets/icon/favicon-16x16.png">
    <link rel="mask-icon" href="/assets/icon/safari-pinned-tab.svg" color="#bf0c0c">
    <link rel="shortcut icon" href="/assets/icon/favicon.ico">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="msapplication-config" content="browserconfig.xml">
    <meta name="theme-color" content="#ffffff">
    <meta name="format-detection" content="telephone=no">
    <meta name="msapplication-tap-highlight" content="no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-title" content="Seleto Food">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <meta http-equiv="x-ua-compatible" content="IE=Edge">
    <link rel="apple-touch-icon" href="/assets/icon/icon.png">
    <link rel="icon" type="image/x-icon" href="/assets/icon/favicon.ico">
</head>
<body>
    <app-root></app-root>
</body>
</html>
```

This is what is being compiled into www folder after `npm run build`:

```html

  | <!DOCTYPE html><html dir="ltr"><head></head><body> <meta charset="utf-8"> <title>Seleto Food</title> <meta name="viewport" content="viewport-fit=cover, width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no"> <link rel="apple-touch-icon" sizes="180x180" href="/assets/icon/apple-touch-icon.png"> <link rel="icon" type="image/png" sizes="192x192" href="/assets/icon/android-chrome-192x192.png"> <link rel="icon" type="image/png" sizes="512x512" href="/assets/icon/android-chrome-512x512.png"> <link rel="icon" type="image/png" sizes="32x32" href="/assets/icon/favicon-32x32.png"> <link rel="icon" type="image/png" sizes="16x16" href="/assets/icon/favicon-16x16.png"> <link rel="mask-icon" href="/assets/icon/safari-pinned-tab.svg" color="#bf0c0c"> <link rel="shortcut icon" href="/assets/icon/favicon.ico"> <meta name="msapplication-TileColor" content="#ffffff"> <meta name="msapplication-config" content="browserconfig.xml"> <meta name="theme-color" content="#ffffff"> <meta name="format-detection" content="telephone=no"> <meta name="msapplication-tap-highlight" content="no"> <meta name="apple-mobile-web-app-capable" content="yes"> <meta name="apple-mobile-web-app-title" content="Seleto Food"> <meta name="apple-mobile-web-app-status-bar-style" content="black"> <meta http-equiv="x-ua-compatible" content="IE=Edge"> <link href="/build/app.css" rel="stylesheet"> <script data-resources-url="/build/app/">!function(e,t,n,o,r,i,s,a,p,c,u,l,d,h){for(u=e.App=e.App\|\|{},(l=t.createElement("style")).innerHTML=p+"{visibility:hidden}.hydrated{visibility:inherit}",l.setAttribute("data-styles",""),d=t.head.querySelector("meta[charset]"),t.head.insertBefore(l,d?d.nextSibling:t.head.firstChild),function(e,t,n){(e["s-apps"]=e["s-apps"]\|\|[]).push("App"),n.componentOnReady\|\|(n.componentOnReady=function(){var t=this;function n(n){if(t.nodeName.indexOf("-")>0){for(var o=e["s-apps"],r=0,i=0;i<o.length;i++)if(e[o[i]].componentOnReady){if(e[o[i]].componentOnReady(t,n))return;r++}if(r<o.length)return void(e["s-cr"]=e["s-cr"]\|\|[]).push([t,n])}n(null)}return e.Promise?new e.Promise(n):{then:n}})}(e,0,c),r=r\|\|u.resourcesUrl,l=(d=t.querySelectorAll("script")).length-1;l>=0&&!(h=d[l]).src&&!h.hasAttribute("data-resources-url");l--);d=h.getAttribute("data-resources-url"),!r&&d&&(r=d),!r&&h.src&&(r=(d=h.src.split("/").slice(0,-1)).join("/")+(d.length?"/":"")+"app/"),l=t.createElement("script"),function(e,t,n,o){return!(t.search.indexOf("core=esm")>0)&&(!(!(t.search.indexOf("core=es5")>0\|\|"file:"===t.protocol)&&e.customElements&&e.customElements.define&&e.fetch&&e.CSS&&e.CSS.supports&&e.CSS.supports("color","var(--c)")&&"noModule"in n)\|\|function(e){try{return new Function('import("")'),!1}catch(e){}return!0}())}(e,e.location,l)?l.src=r+"app.pbzuur0c.js":(l.src=r+"app.yvtjddup.js",l.setAttribute("type","module"),l.setAttribute("crossorigin",!0)),l.setAttribute("data-resources-url",r),l.setAttribute("data-namespace","app"),t.head.appendChild(l)}(window,document,0,0,0,0,0,0,"app-home,app-profile,app-root,ion-animation-controller,ion-app,ion-back-button,ion-button,ion-buttons,ion-content,ion-header,ion-icon,ion-item,ion-label,ion-nav,ion-ripple-effect,ion-route,ion-router,ion-title,ion-toast,ion-toast-controller,ion-toggle,ion-toolbar",HTMLElement.prototype);</script> <link rel="apple-touch-icon" href="/assets/icon/icon.png"> <link rel="icon" type="image/x-icon" href="/assets/icon/favicon.ico"> <link rel="manifest" href="/manifest.json"> <app-root></app-root> </body></html>
-- | --
```

**Comment by JCKodel** [comment]
Tried to redo my project step by step and I found the cause for this bug:

Whenever I ask Visual Studio to save my index.html file (with the original `npm init stencil` contents), the `npm run build` corrupts it.

Usually, I save my files in UTF-8 with signature (Codepage 65001). Saving them with Wester European (Windows) - Codepage 1252 works fine (but accented characters like ç, á, ã, etc. won't display correctly in the browser).

Unicode (UTF-8 without signature) - Codepage 65001 also works fine.

So, basically: UTF-8 BOM causes Stencil builder to corrupt the index.html file while injecting things in it.
